### ISSUE TITLE
C# deploy example 报错：System.AccessViolationException

### ISSUE THREAD

**Issue by carychai** [PaddlePaddle/PaddleX#1185]
问题类型：模型部署
**问题描述**  
模型是paddlex的示例果蔬分类训练出来的
========================  
![3333](https://user-images.githubusercontent.com/26074835/136347212-b5b260d0-0757-4a73-8653-b21a5f43a9e2.png)

**Comment by FlyingQianMM** [comment]
根据报错信息来看，正在读写一块被保护的内存，请问下是否有修改代码？

**Comment by carychai** [comment]
没有修改代码
















在 2021-10-08 10:47:39，"FlyingQianMM" ***@***.***> 写道：

根据报错信息来看，正在读写一块被保护的内存，请问下是否有修改代码？

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by FlyingQianMM** [comment]
看你选择的平台是Any CPU，我们给出的教程是选择x64，https://github.com/PaddlePaddle/PaddleX/tree/develop/examples/C%23_deploy

另外不使用c#部署，直接使用c++部署可以运行么？

**Comment by FlyingQianMM** [comment]
另外，目前有用多线程调用么？麻烦把使用情况描述的更清楚 一些

**Comment by carychai** [comment]
> 看你选择的平台是Any CPU，我们给出的教程是选择x64，https://github.com/PaddlePaddle/PaddleX/tree/develop/examples/C%23_deploy
> 
> 另外不使用c#部署，直接使用c++部署可以运行么？

我改成x64，报一样的错误

**Comment by carychai** [comment]
> 另外，目前有用多线程调用么？麻烦把使用情况描述的更清楚 一些
![4444](https://user-images.githubusercontent.com/26074835/136509389-4e928b34-a5e1-4032-81f7-cf07cd27d543.png)
用的是提供的C# demo来推理的，C#和model_infer c++代码都没修改过;

**Comment by FlyingQianMM** [comment]
> 另外不使用c#部署，直接使用c++部署可以运行么？

用c++demo而不是c#demo可以正常运行么？

**Comment by carychai** [comment]
> > 另外不使用c#部署，直接使用c++部署可以运行么？
> 
> 用c++demo而不是c#demo可以正常运行么？

如何操作？有说明文档吗？

**Comment by FlyingQianMM** [comment]
这里哈 https://github.com/PaddlePaddle/PaddleX/tree/develop/deploy/cpp

**Comment by carychai** [comment]
> 这里哈 https://github.com/PaddlePaddle/PaddleX/tree/develop/deploy/cpp

生成model_infer.exe 可正常执行推理，如下：

![5555](https://user-images.githubusercontent.com/26074835/137050400-68e742aa-8da9-4562-87e0-e9ad9241baa7.png)


-----Log--
D:\PaddleX-develop\deploy\cpp\build_out\paddle_deploy\Release>model_infer.exe --model_filename=e:\paddlex\P0004-T0004_export_model\inference_model\inference_model\model.pdmodel --params_filename=e:\PaddleX\P0004-T0004_export_model\inference_model\inference_model\model.pdiparams --cfg_file=e:\PaddleX\P0004-T0004_export_model\inference_model\inference_model\model.yml --image=e:\PaddleX\datasets\D0004\xihongshi\1.jpg --model_type=paddlex
REGISTER_CLASS:paddlex
init PaddleXModel,model_type=paddlex
WARNING: Logging before InitGoogleLogging() is written to STDERR
I1013 09:17:48.576020 20072 analysis_predictor.cc:155] Profiler is deactivated, and no profiling report will be generated.
I1013 09:17:48.625818 20072 analysis_predictor.cc:553] MKLDNN is enabled
e[1me[35m--- Running analysis [ir_graph_build_pass]e[0m
e[1me[35m--- Running analysis [ir_graph_clean_pass]e[0m
e[1me[35m--- Running analysis [ir_analysis_pass]e[0m
e[32m--- Running IR pass [mkldnn_placement_pass]e[0m
e[32m--- Running IR pass [simplify_with_basic_ops_pass]e[0m
e[32m--- Running IR pass [attention_lstm_fuse_pass]e[0m
e[32m--- Running IR pass [seqconv_eltadd_relu_fuse_pass]e[0m
e[32m--- Running IR pass [seqpool_cvm_concat_fuse_pass]e[0m
e[32m--- Running IR pass [mul_lstm_fuse_pass]e[0m
e[32m--- Running IR pass [fc_gru_fuse_pass]e[0m
e[32m--- Running IR pass [mul_gru_fuse_pass]e[0m
e[32m--- Running IR pass [seq_concat_fc_fuse_pass]e[0m
e[32m--- Running IR pass [squeeze2_matmul_fuse_pass]e[0m
e[32m--- Running IR pass [reshape2_matmul_fuse_pass]e[0m
e[32m--- Running IR pass [flatten2_matmul_fuse_pass]e[0m
e[32m--- Running IR pass [map_matmul_to_mul_pass]e[0m
I1013 09:17:48.764482 20072 graph_pattern_detector.cc:101] ---  detected 1 subgraphs
e[32m--- Running IR pass [fc_fuse_pass]e[0m
I1013 09:17:48.771344 20072 graph_pattern_detector.cc:101] ---  detected 1 subgraphs
e[32m--- Running IR pass [repeated_fc_relu_fuse_pass]e[0m
e[32m--- Running IR pass [squared_mat_sub_fuse_pass]e[0m
e[32m--- Running IR pass [conv_bn_fuse_pass]e[0m
I1013 09:17:48.897234 20072 graph_pattern_detector.cc:101] ---  detected 36 subgraphs
e[32m--- Running IR pass [conv_eltwiseadd_bn_fuse_pass]e[0m
e[32m--- Running IR pass [conv_transpose_bn_fuse_pass]e[0m
e[32m--- Running IR pass [conv_transpose_eltwiseadd_bn_fuse_pass]e[0m
e[32m--- Running IR pass [is_test_pass]e[0m
e[32m--- Running IR pass [runtime_context_cache_pass]e[0m
e[32m--- Running IR pass [depthwise_conv_mkldnn_pass]e[0m
I1013 09:17:48.940842 20072 graph_pattern_detector.cc:101] ---  detected 17 subgraphs
e[32m--- Running IR pass [conv_bn_fuse_pass]e[0m
I1013 09:17:48.972088 20072 graph_pattern_detector.cc:101] ---  detected 17 subgraphs
e[32m--- Running IR pass [conv_eltwiseadd_bn_fuse_pass]e[0m
e[32m--- Running IR pass [conv_affine_channel_fuse_pass]e[0m
e[32m--- Running IR pass [conv_eltwiseadd_affine_channel_fuse_pass]e[0m
e[32m--- Running IR pass [conv_transpose_bn_fuse_pass]e[0m
e[32m--- Running IR pass [conv_transpose_eltwiseadd_bn_fuse_pass]e[0m
e[32m--- Running IR pass [conv_bias_mkldnn_fuse_pass]e[0m
e[32m--- Running IR pass [conv_transpose_bias_mkldnn_fuse_pass]e[0m
e[32m--- Running IR pass [conv3d_bias_mkldnn_fuse_pass]e[0m
e[32m--- Running IR pass [conv_elementwise_add_mkldnn_fuse_pass]e[0m
I1013 09:17:49.085880 20072 graph_pattern_detector.cc:101] ---  detected 10 subgraphs
I1013 09:17:49.090762 20072 conv_elementwise_add_mkldnn_fuse_pass.cc:338] Fused graph 10
e[32m--- Running IR pass [conv_concat_relu_mkldnn_fuse_pass]e[0m
e[32m--- Running IR pass [conv_relu_mkldnn_fuse_pass]e[0m
e[32m--- Running IR pass [conv_leaky_relu_mkldnn_fuse_pass]e[0m
e[32m--- Running IR pass [conv_relu6_mkldnn_fuse_pass]e[0m
I1013 09:17:49.134243 20072 graph_pattern_detector.cc:101] ---  detected 36 subgraphs
e[32m--- Running IR pass [conv_swish_mkldnn_fuse_pass]e[0m
e[32m--- Running IR pass [conv_hard_swish_mkldnn_fuse_pass]e[0m
e[32m--- Running IR pass [scale_matmul_fuse_pass]e[0m
e[37m---    fused 0 scale with matmule[0m
e[32m--- Running IR pass [reshape_transpose_matmul_mkldnn_fuse_pass]e[0m
e[37m---    Fused 0 ReshapeTransposeMatmulMkldnn patternse[0m
e[37m---    Fused 0 ReshapeTransposeMatmulMkldnn patterns with transpose's xshapee[0m
e[37m---    Fused 0 ReshapeTransposeMatmulMkldnn patterns with reshape's xshapee[0m
e[37m---    Fused 0 ReshapeTransposeMatmulMkldnn patterns with reshape's xshape with transpose's xshapee[0m
e[32m--- Running IR pass [matmul_transpose_reshape_fuse_pass]e[0m
e[37m---    Fused 0 MatmulTransposeReshape patternse[0m
e[32m--- Running IR pass [batch_norm_act_fuse_pass]e[0m
e[37m---    fused 0 batch norm with relu activatione[0m
e[1me[35m--- Running analysis [ir_params_sync_among_devices_pass]e[0m
e[1me[35m--- Running analysis [adjust_cudnn_workspace_size_pass]e[0m
e[1me[35m--- Running analysis [inference_op_replace_pass]e[0m
e[1me[35m--- Running analysis [ir_graph_to_program_pass]e[0m
I1013 09:17:49.297842 20072 analysis_predictor.cc:598] ======= optimize end =======
I1013 09:17:49.298820 20072 naive_executor.cc:107] ---  skip [feed], feed -> image
I1013 09:17:49.302733 20072 naive_executor.cc:107] ---  skip [save_infer_model/scale_0.tmp_1], fetch -> fetch
I1013 09:17:49.336253 20072 device_context.cc:519] oneDNN v2.2.0
Classify(4      xihongshi       0.99999893)

**Comment by cjh3020889729** [comment]
可以发一下debug或者release下的bin文件夹中的dll截图吗？确认下是否有缺少dll

**Comment by carychai** [comment]
> 可以发一下debug或者release下的bin文件夹中的dll截图吗？确认下是否有缺少dll
![6666](https://user-images.githubusercontent.com/26074835/137070926-2a365ced-d233-440f-9e56-6fbb3ce98ab7.png)

**Comment by cjh3020889729** [comment]
选择可视化界面中推理类型为paddlex才能执行paddlex训练的模型
